src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -52,18 +53,19 @@ public static void ReadAndWriteUntilPosition(BinaryReader reader, BinaryWriter w
                 throw new ArgumentOutOfRangeException(nameof(position), Strings.SignedPackageArchiveIOInvalidRead);
             }
 
-            while (reader.BaseStream.Position + _bufferSize < position)
-            {
-                var bytes = reader.ReadBytes(_bufferSize);
-                writer.Write(bytes);
-            }
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(_bufferSize);
```

Comment:
The intention is to write all bytes up to but not including `position` from the `reader` to the `writer` and I maintained this, and just simplified this into a single pass.

While its also tempting to unify all three versions of the method below into a single method taking a func, I avoided it to avoid the capture/delegate allocations.